core-backend/src/runtime.rs

```diff
@@ -184,4 +164,19 @@ where
             },
         }
     }
+
+    pub fn check_func_forbiddenness(&mut self, syscall_name: SyscallName) -> Result<(), HostError> {
+        let endpoint_forbidden_funcs = self.ext_mut().endpoint_dispatch_kind().forbidden_funcs();
```

Comment:
maybe inverse this check into ?
entry.forbids(call) -> bool

Reply:
Yeah, that seems like a more direct solution